### PR TITLE
Feature/12517 fss UI simplified search add download all button

### PR DESCRIPTION
**Pull request by mohamme15315**
Added download all button on simplified search result with unit testing

### CODE REVIEW

File: Tests/UnitTests/fss-search-results.component.spec.ts [hunk 1]
```diff
@@ -135,100 +139,197 @@ describe('FssSearchResultsComponent', () => {
...
+    var batches = component.searchResult[0];
+    component.ngOnChanges();
+    tick(100);
+    fixture.detectChanges();
+
+    fixture.whenStable().then(() => {
+      batches.forEach((item: SearchResultViewModel) => {
+        let css = "[id='" + item.BatchID.value + "']";
+        const home = fixture.debugElement.query(By.css(css)).nativeElement;
+        if (item.allFilesZipSize != undefined && item.allFilesZipSize != null) {
```
Comment (crdandoukho) [inline Tests/UnitTests/fss-search-results.component.spec.ts:178]: Please change this to a simple 'truthy' condition
```suggestion
        if (item.allFilesZipSize) {
```
If `allFilesZipSize` is zero we don't want to display the download button

Comment (crdandoukho) [inline Tests/UnitTests/fss-search-results.component.spec.ts:178]: Please add an inverse assertion to prove the button is present when `allFileZipSize` is truthy

File: src/app/features/fss-search/fss-search-results/fss-search-results.component.html [hunk 2]
```diff
@@ -41,11 +42,23 @@
...
+            <div class="fileDownload" [attr.rel]="element.FileLink" tabindex="0" role="button"
+              attr.aria-label="Download File for {{element.FileName}}"><i class="fa fa-download fa-1x"></i></div>
+          </td>
         </ng-container>
         <tr ukho-header-row *ukhoHeaderRowDef="batch.batchFileDetails.columnData"></tr>
         <tr ukho-row *ukhoRowDef="let row; columns: batch.batchFileDetails.columnData"></tr>
       </table>
+
+
+      <ng-container *ngIf="batch.allFilesZipSize!=null">
```
Comment (crdandoukho) [inline src/app/features/fss-search/fss-search-results/fss-search-results.component.html:55]: Please change this to a simple 'truthy' condition
```suggestion
      <ng-container *ngIf="batch.allFilesZipSize">
```
Doing this covers undefined and zero

File: src/app/features/fss-search/fss-search-results/fss-search-results.component.ts [hunk 3]
```diff
@@ -28,7 +28,8 @@ export class FssSearchResultsComponent implements OnChanges {
           batchFileDetails: this.getBatchFileDetails(batches[i]),
           BatchID: { key: 'Batch ID', value: batches[i]['batchId'] },
           BatchPublishedDate: { key: 'Batch published date', value: batches[i]['batchPublishedDate'] },
-          ExpiryDate: { key: 'Batch expiry date', value: batches[i]['expiryDate'] }
+          ExpiryDate: { key: 'Batch expiry date', value: batches[i]['expiryDate'] },
+          allFilesZipSize:batches[i]["allFilesZipSize"]
```
Comment (crdandoukho) [inline src/app/features/fss-search/fss-search-results/fss-search-results.component.ts:32]: Please use single quote delimiter
```suggestion
          allFilesZipSize:batches[i]['allFilesZipSize']
```

**Review by crdandoukho — CHANGES_REQUESTED** [review]
In addition to my comments, please also fix the failing unit test:
`FssSearchResultsComponent › should return search result count 1 when search result for 1 batch is provided` in (fss-search-results.component.spec.ts)

At the time of writing I have not been able to run the functional tests because the DevOps pipeline is running very slowly

File: Tests/UnitTests/fss-search-results.component.spec.ts [hunk 4]
```diff
@@ -135,100 +139,249 @@ describe('FssSearchResultsComponent', () => {
...
+    component.ngOnChanges();
+    tick(100);
+    fixture.detectChanges();
+    fixture.whenStable().then(() => {
+      batches.forEach((item: SearchResultViewModel) => {
+        let css = "[id='" + item.BatchID.value + "']";
+        const home = fixture.debugElement.query(By.css(css)).nativeElement;
+        if (item.allFilesZipSize) {
+          expect(home.querySelector("ukho-button").textContent).toEqual("Download all");
+        }
```
Comment (crdandoukho) [inline Tests/UnitTests/fss-search-results.component.spec.ts:179]: Please extend this test to prove the [Download all] button is _not_ displayed when `allFilesZipSize` is falsy

For example
```suggestion
        } else {
         assert here that [Download all] button is not displayed
        }
```

**Review by crdandoukho — CHANGES_REQUESTED** [review]
Please extend the new test to prove the [Download all] button is _not_ displayed when `allFilesZipSize` is falsy

**Review by bhalcha13813 — APPROVED** [review]
LGTM

**Pull request merged**